include/maths/CBoostedTreeFactory.h

```diff
@@ -139,8 +139,10 @@ class MATHS_EXPORT CBoostedTreeFactory final {
     //! Estimate the maximum booking memory that training the boosted tree on a
     //! data frame with \p numberRows row and \p numberColumns columns will use.
     std::size_t estimateMemoryUsage(std::size_t numberRows, std::size_t numberColumns) const;
-    //! Get the number of columns training the model will add to the data frame.
-    std::size_t numberExtraColumnsForTrain() const;
+    //! Estimate the number of columns training the model will add to the data frame.
+    static std::size_t estimatedExtraColumns(std::size_t numberColumns,
```

Comment:
Don't we know the number of extra columns for sure? "Estimate" sounds like something that may deviate in the future, e.g. `estimateMemoryUsage`.

Reply:
We don't know exactly how many encoded features we will choose upfront (for example it'll vary with categorical fields present and how much information they carry). Since we now have additional some storage for each such feature this is only approximate.